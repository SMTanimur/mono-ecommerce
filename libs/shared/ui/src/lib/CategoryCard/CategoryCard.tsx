import Image from 'next/image';
import { Button } from '../button';
import { Category } from '@ecommerce/shared/graphql/refactored-types';
import classes from './CategoryCard.module.css';
import clsx from 'clsx';
import Link from 'next/link';
import { getImageURL } from '@ecommerce/shared/utils/url';
import FeatherIcon from 'feather-icons-react';

export interface CategoryCardProps {
  category: Category;
  color?: 'yellow' | 'green' | 'blue';
}

export const CategoryCard: React.FC<CategoryCardProps> = (props) => {
  const { category, color = 'blue' } = props;

  const { smallDescription, image, slug, name } = category;

  return (
    <Link href={`/s?c=${slug}`}>
      <a
        className={clsx(
          'h-full',
          'flex flex-col',
          'hover:(underline)',
          'focus:(underline)'
        )}
      >
        <div
          className={clsx(
            'w-full relative',
            'bg-gray-100',
            classes['category-card-aspect']
          )}
        >
          {image && image.data && (
            <Image
              src={getImageURL(image.data.attributes?.url)}
              alt=""
              layout="fill"
              objectFit="cover"
            />
          )}
        </div>
        <div
          className={clsx(
            'p-5',
            'flex flex-col flex-1 gap-4',
            color === 'blue' && 'bg-blue-400',
            color === 'green' && 'bg-green-400',
            color === 'yellow' && 'bg-yellow-400',
            'sm:(p-8 gap-8)'
          )}
        >
          <h3
            className={clsx(
              'font-bold text-xl leading-relaxed',
              'sm:(text-2xl)'
            )}
          >
            {smallDescription}
          </h3>
          <Button
            classes="w-max mt-auto"
            tag="button"
            type="black"
            elProps={{ title: `go to ${name} category` }}
          >
            <div
              className={clsx('h-6 w-6 relative', 'sm:(h-8 w-8)')}
              aria-hidden="true"
            >
              <FeatherIcon icon="arrow-right" size={32} />
            </div>
          </Button>
        </div>
      </a>
    </Link>
  );
};
