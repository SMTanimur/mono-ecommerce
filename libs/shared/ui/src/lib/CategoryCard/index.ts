import { CategoryCard, CategoryCardProps } from './CategoryCard';

export { CategoryCard };

export type { CategoryCardProps };
