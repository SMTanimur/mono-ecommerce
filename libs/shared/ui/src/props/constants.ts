import { ButtonProps } from '@ecommerce/shared/ui/Button';
import {CartItemProps} from "@ecommerce/shared/ui/CartItem";
import {ProductSidebarProps} from "@ecommerce/shared/ui/ProductSidebar";
import {CartSidebarProps} from "@ecommerce/shared/ui/CartSidebar";
import {CategoryCardProps} from "@ecommerce/shared/ui/CategoryCard";
import {IconProps} from "@ecommerce/shared/ui/Icon";
import {ItemCardProps} from "@ecommerce/shared/ui/ItemCard";
import {NavButtonProps} from "@ecommerce/shared/ui/NavButton";
import {ProductGalleryProps} from "@ecommerce/shared/ui/ProductGallery";

import cat from './cat.jpg';
import cat2 from './cat2.jpeg';

console.log(cat2)

export const PRODUCT: ProductSidebarProps['product'] = {
  slug: 'some-name',
  price: 65,
  name: 'Some name',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'Small Description',
  description: 'Description',
  information: [
    {
      title: 'Title1',
      content: 'content1',
    },
  ],
};

export const ITEM: ItemCardProps['item'] = {
  slug: 'cat-cat',
  price: 65,
  name: 'cat cat',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'smallDescription',
  description: 'description',
};

export const BUTTON_DEFAULT_PROPS: ButtonProps = {
  children: 'hello',
  tag: 'button',
  size: 'base',
  type: 'primary',
  round: 'full',
};

export const CART_ITEM_DEFAULT_PROPS: CartItemProps = {
  inCartAmount: 1,
  item: PRODUCT,
  handleCartMinusClick: () => {
    console.log('cart minus');
  },
  handleCartPlusClick: () => {
    console.log('cart plus');
  },
  handleRemoveFromCart: () => {
    console.log('cart remove');
  },
  toggleFavorite: () => {
    console.log('favorite');
  },
};

export const CART_SIDEBAR_DEFAULT_PROPS: CartSidebarProps = {
  cartTotal: 100,
  isUser: true,
  handleCheckoutClick: async () => {
    console.log('checkout');
  },
  isLoading: false,
};

export const CATEGORY_CARD_DEFAULT_PROPS: CategoryCardProps = {
  category: {
    smallDescription:
      'smalll des',
    image: {
      data: {
        attributes: {
          url: cat,
        },
      },
    },
    slug: '1',
  },
};
export const ICON_DEFAULT_PROPS: IconProps = {
  icon: 'cart',
};

export const ITEM_CARD_DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleCartClick: () => console.log('cart clicked'),
  handleAmountCartClick: () => console.log('amount cart clicked'),
  handleFavoriteClick: () => console.log('favorite clicked'),
};
export const NAV_BUTTON_DEFAULT_PROPS: NavButtonProps = {
  type: 'heart',
};
export const PRODUCT_GALLERY_DEFAULT_PROPS: ProductGalleryProps = {
  images: [
    {
      attributes: {
        url: cat,
      },
    },
    {
      attributes: {
        url: cat2,
      },
    },
  ],
};
