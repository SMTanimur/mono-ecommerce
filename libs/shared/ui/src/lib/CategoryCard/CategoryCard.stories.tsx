import { Meta, Story } from '@storybook/react';

import { CATEGORY_CARD_DEFAULT_PROPS } from '@ecommerce/shared/props/constants';

import { CategoryCard, CategoryCardProps } from './CategoryCard';

export default {
  component: CategoryCard,
  title: 'Components/CategoryCard',
} as Meta;

const Template: Story<CategoryCardProps> = (args) => {
  return (
    <div className="w-96">
      <CategoryCard {...args} />
    </div>
  );
};

export const Primary = Template.bind({});
Primary.args = CATEGORY_CARD_DEFAULT_PROPS;
