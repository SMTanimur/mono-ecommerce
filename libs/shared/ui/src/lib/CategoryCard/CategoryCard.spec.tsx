import { CATEGORY_CARD_DEFAULT_PROPS } from '../../props/constants';
import { render, RenderResult } from '@testing-library/react';
import {create} from 'react-test-renderer';

import { CategoryCard } from './CategoryCard';

describe('CategoryCard', () => {
  let renderResult: RenderResult;
  const props = CATEGORY_CARD_DEFAULT_PROPS;

  beforeEach(() => {
    renderResult = render(<CategoryCard {...props} />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;

    const tree = create(<CategoryCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();

    expect(baseElement).toBeTruthy();
  });
});
