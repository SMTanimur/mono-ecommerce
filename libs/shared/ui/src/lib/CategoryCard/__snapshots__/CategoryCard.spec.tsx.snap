// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard should render successfully 1`] = `
<a
  className="h-full flex flex-col hover:(underline) focus:(underline)"
  href="/s?c=1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="w-full relative bg-gray-100 category-card-aspect"
  >
    Next image stub
  </div>
  <div
    className="p-5 flex flex-col flex-1 gap-4 bg-blue-400 sm:(p-8 gap-8)"
  >
    <h3
      className="font-bold text-xl leading-relaxed sm:(text-2xl)"
    >
      smalll des
    </h3>
    <button
      className="inline-block cursor-pointer transform-gpu transition-all active:scale-92 rounded-full p-3 bg-black text-white hover:(bg-dark-400) w-max mt-auto"
      title="go to undefined category"
    >
      <div
        aria-hidden="true"
        className="h-6 w-6 relative sm:(h-8 w-8)"
      >
        <svg
          className="feather feather-arrow-right "
          fill="none"
          height={32}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={32}
        >
          <g
            dangerouslySetInnerHTML={
              Object {
                "__html": "<line x1=\\"5\\" y1=\\"12\\" x2=\\"19\\" y2=\\"12\\"></line><polyline points=\\"12 5 19 12 12 19\\"></polyline>",
              }
            }
          />
        </svg>
      </div>
    </button>
  </div>
</a>
`;
